### ISSUE TITLE
Error when runing atop

### ISSUE THREAD

**Issue by cornerot** [elves/elvish#336]
I got this error when start atop:

`Exception: atop stopped by signal stopped (tty input) (pid=16175)`

And then:

```
Unbound: Ctrl-M
Unbound: Ctrl-M
Unbound: Ctrl-M
```

**Comment by xiaq** [comment]
I cannot reproduce the issue. Did you run `atop` directly from the (elvish) command line?

**Comment by cornerot** [comment]
https://asciinema.org/a/2ridm1p39bibphb8ji10if2s5

**Comment by xiaq** [comment]
Hmm, I cannot reproduce this. :-/

**Comment by cornerot** [comment]
I can reproduce this in the Arch linux, but cannot in the Centos 7

**Comment by huiyiqun** [comment]
Still not reproducible on Arch Linux.

**Comment by therealpxc** [comment]
I can reproduce on NixOS. The ‘Unbound: Ctrl-M’ messages appear whenever I hit enter.

Maybe you guys can get the Nix packages to reproduce the bad behavior even on other distros. If you [install Nix](http://nixos.org/nix/), you can try this using our elvish and atop packages with

`nix-env -i elvish atop`

That will install these versions of elvish and atop alongside your system ones. You can remove them with

    nix-env -e elvish
    nix-env -e atop

or uninstall Nix entirely with a simple

`rm -rf /nix`

**Comment by cornerot** [comment]
I cannot reproduce in the Arch after last system update.

**Comment by HeavyHorst** [comment]
I have the same issue with the python shell when i suspend the shell with ctrl+z :
I'm using fedora 25.

```
~> python
Python 2.7.13 (default, May 10 2017, 20:04:28)
[GCC 6.3.1 20161221 (Red Hat 6.3.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> Exception: python stopped by signal stopped (pid=18839)
Traceback:
  [interactive], line 1:
    python
Unbound: Ctrl-M
Unbound: Ctrl-M
Unbound: Ctrl-M
Unbound: Ctrl-M
Unbound: Ctrl-M
Unbound: Ctrl-M
Unbound: Ctrl-M

```

**Comment by HeavyHorst** [comment]
This is actually the same issue as #399 

The -icrnl setting (icrnl means “turn carriage returns into newlines”) is not cleaned from the terminal settings when the process is stopped. After running `stty sane` , evrything works again.

**Comment by HeavyHorst** [comment]
I have also tested it with zsh and zsh also doesn't reset the settings. However, zsh handles ctrl + m and \ n equally.

```
rkaufmann@chefkaufmann ~ % stty -a
speed 38400 baud; rows 58; columns 239; line = 0;
intr = ^C; quit = ^\; erase = ^?; kill = ^U; eof = ^D; eol = <undef>; eol2 = <undef>; swtch = <undef>; start = ^Q; stop = ^S; susp = ^Z; rprnt = ^R; werase = ^W; lnext = ^V; discard = ^O; min = 1; time = 0;
-parenb -parodd -cmspar cs8 -hupcl -cstopb cread -clocal -crtscts
-ignbrk -brkint -ignpar -parmrk -inpck -istrip -inlcr -igncr icrnl ixon -ixoff -iuclc -ixany -imaxbel iutf8
opost -olcuc -ocrnl onlcr -onocr -onlret -ofill -ofdel nl0 cr0 tab0 bs0 vt0 ff0
isig icanon iexten echo echoe echok -echonl -noflsh -xcase -tostop -echoprt echoctl echoke -flusho -extproc
rkaufmann@chefkaufmann ~ % python
Python 2.7.13 (default, May 10 2017, 20:04:28)
[GCC 6.3.1 20161221 (Red Hat 6.3.1-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>>
[1]  + 10511 suspended  python
148 rkaufmann@chefkaufmann ~ % stty -a                                                                                                                                                                                                      :(
speed 38400 baud; rows 58; columns 239; line = 0;
intr = ^C; quit = ^\; erase = ^?; kill = ^U; eof = ^D; eol = <undef>; eol2 = <undef>; swtch = <undef>; start = ^Q; stop = ^S; susp = ^Z; rprnt = ^R; werase = ^W; lnext = <undef>; discard = ^O; min = 1; time = 0;
-parenb -parodd -cmspar cs8 -hupcl -cstopb cread -clocal -crtscts
-ignbrk -brkint -ignpar -parmrk -inpck -istrip -inlcr -igncr -icrnl ixon -ixoff -iuclc -ixany -imaxbel iutf8
opost -olcuc -ocrnl onlcr -onocr -onlret -ofill -ofdel nl0 cr0 tab0 bs0 vt0 ff0
isig icanon iexten echo echoe echok -echonl -noflsh -xcase -tostop -echoprt echoctl echoke -flusho -extproc
```

**Comment by xiaq** [comment]
I suppose this has been fixed when #399 was fixe.
